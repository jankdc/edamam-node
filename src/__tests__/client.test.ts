import * as got from "got";

import {
  DietLabel,
  EdamamClient,
  HealthLabel,
  NutrientCode,
} from "../";

describe("EdamamClient", () => {
  let client: EdamamClient;
  let getSpy: jest.SpyInstance;

  beforeEach(() => {
    client = new EdamamClient({
      appId: "some-id",
      appKey: "some-key",
      baseUrl: "https://some-url.com",
    });
    getSpy = jest.spyOn(got, "get").mockResolvedValue({body: {}} as any);
  });

  afterEach(() => {
    getSpy.mockClear();
  });

  describe("searchRecipesByQuery", () => {
    it("requests correctly (no options)", async () => {
      await client.searchRecipesByQuery("some-query");
      expect(getSpy.mock.calls[0][0]).toMatchSnapshot();
    });

    it("requests correctly (from, to)", async () => {
      await client.searchRecipesByQuery("some-query", {
        from: 1234,
        to: 4321,
      });
      expect(getSpy.mock.calls[0][0]).toMatchSnapshot();
    });

    it("requests correctly (diet)", async () => {
      await client.searchRecipesByQuery("some-query", {
        diet: DietLabel.Balanced,
      });
      expect(getSpy.mock.calls[0][0]).toMatchSnapshot();
    });

    it("requests correctly (ingr)", async () => {
      await client.searchRecipesByQuery("some-query", {
        ingr: 1234,
      });
      expect(getSpy.mock.calls[0][0]).toMatchSnapshot();
    });

    it("requests correctly (time)", async () => {
      await client.searchRecipesByQuery("some-query", {
        time: "123+",
      });
      expect(getSpy.mock.calls[0][0]).toMatchSnapshot();
    });

    it("requests correctly (health)", async () => {
      await client.searchRecipesByQuery("some-query", {
        health: [HealthLabel.AlcoholFree, HealthLabel.Dairy],
      });
      expect(getSpy.mock.calls[0][0]).toMatchSnapshot();
    });

    it("requests correctly (excluded)", async () => {
      await client.searchRecipesByQuery("some-query", {
        excluded: ["some-excluded"],
      });
      expect(getSpy.mock.calls[0][0]).toMatchSnapshot();
    });

    it("requests correctly (calories)", async () => {
      await client.searchRecipesByQuery("some-query", {
        calories: "some-calories",
      });
      expect(getSpy.mock.calls[0][0]).toMatchSnapshot();
    });

    it("requests correctly (nutrients)", async () => {
      const nutrientsMap = new Map<NutrientCode, string>();
      nutrientsMap.set(NutrientCode.Calcium, "1341+");

      await client.searchRecipesByQuery("some-query", {
        nutrients: nutrientsMap,
      });
      expect(getSpy.mock.calls[0][0]).toMatchSnapshot();
    });
  });

  describe("searchRecipesById", () => {
    it("requests API correctly (no options)", async () => {
      await client.searchRecipesById("some-r");
      expect(getSpy.mock.calls[0][0]).toMatchSnapshot();
    });
  });
});
