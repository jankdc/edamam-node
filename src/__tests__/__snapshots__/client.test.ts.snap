// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EdamamClient searchRecipesById requests API correctly (no options) 1`] = `"https://some-url.com/search?app_id=some-id&app_key=some-key&r=some-r"`;

exports[`EdamamClient searchRecipesByQuery requests correctly (calories) 1`] = `"https://some-url.com/search?calories=some-calories&app_id=some-id&app_key=some-key&q=some-query"`;

exports[`EdamamClient searchRecipesByQuery requests correctly (diet) 1`] = `"https://some-url.com/search?diet=balanced&app_id=some-id&app_key=some-key&q=some-query"`;

exports[`EdamamClient searchRecipesByQuery requests correctly (excluded) 1`] = `"https://some-url.com/search?app_id=some-id&app_key=some-key&q=some-query&excluded=some-excluded"`;

exports[`EdamamClient searchRecipesByQuery requests correctly (from, to) 1`] = `"https://some-url.com/search?from=1234&to=4321&app_id=some-id&app_key=some-key&q=some-query"`;

exports[`EdamamClient searchRecipesByQuery requests correctly (health) 1`] = `"https://some-url.com/search?health=alcohol-free,dairy-free&app_id=some-id&app_key=some-key&q=some-query"`;

exports[`EdamamClient searchRecipesByQuery requests correctly (ingr) 1`] = `"https://some-url.com/search?ingr=1234&app_id=some-id&app_key=some-key&q=some-query"`;

exports[`EdamamClient searchRecipesByQuery requests correctly (no options) 1`] = `"https://some-url.com/search?app_id=some-id&app_key=some-key&q=some-query"`;

exports[`EdamamClient searchRecipesByQuery requests correctly (nutrients) 1`] = `"https://some-url.com/search?app_id=some-id&app_key=some-key&q=some-query&nutrients%5BCA%5D=1341+"`;

exports[`EdamamClient searchRecipesByQuery requests correctly (time) 1`] = `"https://some-url.com/search?time=123+&app_id=some-id&app_key=some-key&q=some-query"`;
